### PR TITLE
Adds rails-7-1 support 

### PR DESCRIPTION
**Pull request by eis-ioki**
In Rails 7.1 ::ActiveRecord::SchemaMigration.table_name is deprecated

### CODE REVIEW

**Comment by petergoldstein** [comment]
Would love to see this merged and released in a new version.  Perhaps also with Ruby 3.2 added to the CI matrix.

**Comment by petergoldstein** [comment]
@eis-ioki You're going to want to add a Rails 7.1 gemfile and exclude the combination of Rails 7.1 with all Rubies earlier than Ruby 3.0.  Plus you'll need to rebase to pick up the Ruby 3.2 addition to CI.

File: lib/database_cleaner/active_record/base.rb [hunk 1]
```diff
@@ -15,7 +15,11 @@ def self.config_file_location
 
     class Base < DatabaseCleaner::Strategy
       def self.migration_table_name
-        ::ActiveRecord::SchemaMigration.table_name
+        if Gem::Version.new("6.0.0") <= ::ActiveRecord.version
+          ::ActiveRecord::Base.connection.schema_migration.table_name
+        else
+          ::ActiveRecord::SchemaMigration.table_name
```
Comment (mathieujobin) [inline lib/database_cleaner/active_record/base.rb:21]: this no longer works in 7.1

please merge this one instead https://github.com/DatabaseCleaner/database_cleaner-active_record/pull/82

Comment (eis-ioki) [reply lib/database_cleaner/active_record/base.rb:21]: That's what this PR is all about. Your should use `::ActiveRecord::Base.connection.schema_migration.table_name` from 6.0.0 onwards.

**Review by etagwerker — APPROVED** [review]
@eis-ioki This is great. Thanks for pushing this forward! 👍🏼

**Comment by petergoldstein** [comment]
@etagwerker Is there going to be a gem release that includes this change published in the near future?  That would be very helpful.

**Comment by etagwerker** [comment]
@petergoldstein https://mastodon.social/@etagwerker/109880784622263889

**Pull request merged**